src/target/cortexm.c

```diff
@@ -679,7 +679,7 @@ static int dcrsr_regnum(target *t, unsigned reg)
 		return regnum_cortex_m[reg];
 	} else if ((t->target_options & TOPT_FLAVOUR_V7MF) &&
 	           (reg < (sizeof(regnum_cortex_m) +
-	                   sizeof(regnum_cortex_mf) / 4))) {
+	                   sizeof(regnum_cortex_mf)) / 4)) {
```

Comment:
From what this looks like, we were previously generating the wrong number of registers in this expression because only the last sizeof() was being divided by 4 but they all should have been - is this a correct assessment?

Reply:
This condition is just a sanity check, but yes, it had a typo that made it too relaxed.